curriculum/challenges/english/25-front-end-development/review-web-performance/6724e18296fa40173cc2437c.md

```diff
@@ -9,7 +9,137 @@ dashedName: review-web-performance
+- Optimize event handlers. If these handlers run too frequently or perform heavy operations, they can slow down the page and increase INP. The solution for this is debouncing. Debouncing ensures that the function only runs after the user stops typing for a short delay - so for example 300ms. This prevents unnecessary calculations and improves performance.
+
+## How Rendering Works in the Browser
+
+- **How Rendering works**: First the browser parses the HTML and builds the DOM. Next, the browser processes the CSS, constructing the CSS Object Model, or CSSOM. This is another tree structure that dictates how elements should be styled. Finally, the browser paints the pixels to the screen, rendering each element based on the calculated styles and layout. In complex pages, this might involve multiple layers that are composited together to form the final visual output.
+
+## How Performance Impacts Sustainability
+
+- **Background Information**: The internet accounts for around 2% of global carbon emissions—that’s the same as the airline industry! Every byte transferred requires electricity, from data centers to user devices. Larger files and inefficient scripts mean more power consumption. A high-performance website isn't just faster—it also reduces unnecessary processing and energy use.
+
+## Ways to Reduce Page Loading Times
+
+- **Optimize Media Assets**: Large images and videos are common culprits for slow load times. By optimizing these assets, you can significantly speed up your site. This includes things like compressing images, using modern formats like WebP and using lazy loading for assets.
+- **Leverage Browser Caching**: Caching allows browsers to store parts of your website locally, reducing load times for returning visitors.
+- **Minify and Compress Files**: Reducing the size of your files can lead to quicker downloads. This includes reducing the size of transmitted files and minifying CSS and JavaScript files.
+
+## Improving "time to usable"
+
+- **Definition**: "time to usable" is the interval from when a user requests a page to when they can meaningfully interact with it. To improve "time to usable" you can lazy load your asset or minimize render-blocking resources.
+
+## Key Metrics for Measuring Performance
+
+- **First Contentful Paint or FCP**: First Contentful Paint or FCP measures how quickly the first piece of content—text or image—appears on the screen. A good FCP is regarded as a time below 1.8 seconds, and a poor FCP is above 3 seconds. You can check your FCP using Chrome DevTools and checking the performance tab
+- **Total Blocking Time**: This shows how long the main thread is blocked by heavy JavaScript tasks. If Total Blocking Time(TBT) is high, users experience sluggish interactions. To improve TBT, break up long tasks and defer non-essential scripts.
+- **Bounce Rate**: This is the percentage of visitors who leave without interacting. if your site has High bounce rates - it might be because your page is too slow.
```

Comment:
```suggestion
- **Bounce Rate**: This is the percentage of visitors who leave without interacting. If your site has high bounce rates it might be because your page is too slow.
```